### PR TITLE
JAVA-2912: Update com.mongodb.client.MongoClient to extend Closeable

### PR DESCRIPTION
**Pull request by thejonathanma**
Updated the MongoClient class in driver-sync to extend Closeable. 

[Evergreen Patch Build](https://evergreen.mongodb.com/version/5b90113d0305b93c4c5eaf07)

### CODE REVIEW

File: driver-sync/src/main/com/mongodb/client/MongoClient.java [hunk 1]
```diff
@@ -101,7 +102,6 @@
      * @return the list databases iterable interface
      */
     ListDatabasesIterable<Document> listDatabases();
-
```
Comment (jyemin) [inline driver-sync/src/main/com/mongodb/client/MongoClient.java]: This is a judgement call, but generally try to avoid unrelated whitespace changes.  As the commits get bigger, they can distract from the important parts.

Whitespace and other formatting issues can always be done as a separate commit, and don't required a Jira issue.

Comment (thejonathanma) [reply driver-sync/src/main/com/mongodb/client/MongoClient.java]: I think that was by accident. I'll undo that change.
